Mathlib/Order/Bounds/Basic.lean

```diff
@@ -0,0 +1,1645 @@
+#align is_lub.of_subset_of_superset IsLUB.of_subset_of_superset
+
+/-- If `a` is a greatest lower bound for sets `s` and `p`, then it is a greater lower bound for any
+set `t`, `s ⊆ t ⊆ p`. -/
+theorem IsGLB.of_subset_of_superset {s t p : Set α} (hs : IsGLB s a) (hp : IsGLB p a) (hst : s ⊆ t)
+    (htp : t ⊆ p) : IsGLB t a :=
+  hs.dual.of_subset_of_superset hp hst htp
+#align is_glb.of_subset_of_superset IsGLB.of_subset_of_superset
+
+theorem IsLeast.mono (ha : IsLeast s a) (hb : IsLeast t b) (hst : s ⊆ t) : b ≤ a :=
+  hb.2 (hst ha.1)
+#align is_least.mono IsLeast.mono
+
+theorem IsGreatest.mono (ha : IsGreatest s a) (hb : IsGreatest t b) (hst : s ⊆ t) : a ≤ b :=
+  hb.2 (hst ha.1)
+#align is_greatest.mono IsGreatest.mono
+
+theorem IsLUB.mono (ha : IsLUB s a) (hb : IsLUB t b) (hst : s ⊆ t) : a ≤ b :=
+  IsLeast.mono hb ha <| upperBounds_mono_set hst
+#align is_lub.mono IsLUB.mono
+
+theorem IsGLB.mono (ha : IsGLB s a) (hb : IsGLB t b) (hst : s ⊆ t) : b ≤ a :=
+  IsGreatest.mono hb ha <| lowerBounds_mono_set hst
+#align is_glb.mono IsGLB.mono
+
+theorem subset_lowerBounds_upperBounds (s : Set α) : s ⊆ lowerBounds (upperBounds s) :=
+  fun _ hx _ hy => hy hx
+#align subset_lower_bounds_upper_bounds subset_lowerBounds_upperBounds
+
+theorem subset_upperBounds_lowerBounds (s : Set α) : s ⊆ upperBounds (lowerBounds s) :=
+  fun _ hx _ hy => hy hx
+#align subset_upper_bounds_lower_bounds subset_upperBounds_lowerBounds
+
+theorem Set.Nonempty.bddAbove_lowerBounds (hs : s.Nonempty) : BddAbove (lowerBounds s) :=
+  hs.mono (subset_upperBounds_lowerBounds s)
+#align set.nonempty.bdd_above_lower_bounds Set.Nonempty.bddAbove_lowerBounds
+
+theorem Set.Nonempty.bddBelow_upperBounds (hs : s.Nonempty) : BddBelow (upperBounds s) :=
+  hs.mono (subset_lowerBounds_upperBounds s)
+#align set.nonempty.bdd_below_upper_bounds Set.Nonempty.bddBelow_upperBounds
+
+/-!
+### Conversions
+-/
+
+
+theorem IsLeast.isGLB (h : IsLeast s a) : IsGLB s a :=
+  ⟨h.2, fun _ hb => hb h.1⟩
+#align is_least.is_glb IsLeast.isGLB
+
+theorem IsGreatest.isLUB (h : IsGreatest s a) : IsLUB s a :=
+  ⟨h.2, fun _ hb => hb h.1⟩
+#align is_greatest.is_lub IsGreatest.isLUB
+
+theorem IsLUB.upperBounds_eq (h : IsLUB s a) : upperBounds s = Ici a :=
+  Set.ext fun _ => ⟨fun hb => h.2 hb, fun hb => upperBounds_mono_mem hb h.1⟩
+#align is_lub.upper_bounds_eq IsLUB.upperBounds_eq
+
+theorem IsGLB.lowerBounds_eq (h : IsGLB s a) : lowerBounds s = Iic a :=
+  h.dual.upperBounds_eq
+#align is_glb.lower_bounds_eq IsGLB.lowerBounds_eq
+
+theorem IsLeast.lowerBounds_eq (h : IsLeast s a) : lowerBounds s = Iic a :=
+  h.isGLB.lowerBounds_eq
+#align is_least.lower_bounds_eq IsLeast.lowerBounds_eq
+
+theorem IsGreatest.upperBounds_eq (h : IsGreatest s a) : upperBounds s = Ici a :=
+  h.isLUB.upperBounds_eq
+#align is_greatest.upper_bounds_eq IsGreatest.upperBounds_eq
+
+theorem is_lub_le_iff (h : IsLUB s a) : a ≤ b ↔ b ∈ upperBounds s := by
```

Comment:
I don't think this name is right, but I'm not sure if it should say `isLub_le_iff` or `isLUB_le_iff`. Seems like you did `isLUB` in some places below.